optional.h

```diff
@@ -23,17 +40,21 @@ class Optional {
 	}
 
 	T& get() {
-		assert(m_valid);
+		if (!m_valid) {
+			throw BadOptional();
+		}
 		return *m_value;
 	}
 		
 	const T& get() const {
```

Comment:
Cool! But now we have 2 identical functions (copy&paste). Could we do some code-reuse? (I.e., implement one function and call it from the other one)